### ISSUE TITLE
Please help advice the tips to reproduce the performance in the paper

### ISSUE THREAD

**Issue by EricYang3721** [JinhwiPark/DepthPrompting#3]
This is very good paper, and I tested the model provided which matches the performance on the paper. However, when I try to train the model with given code, I wasn't able to reproduce models with similar performance.

I saw the there's a couple of differences in the paper and the code.
1. Paper mentioned training was 25 epochs and the code is 16. I changed to 30 epochs, and tested the best one.
2. Paper mentioned "The initial learning rate is 2 × 10−3, and then scaled down with coefficients 0.5, 0.1, and 0.05 every 5 epochs after 10th epoch.". In the code, `--decay 10,15,20,25` and `1.0,0.5,0.05,0.001`. I used the one in the code.
3. Paper doesn't mention about batch size, where I used 12 which is the same as the code.

I also found that during the training, the model initialization seems greatly impact the training loss. In some cases, the loss dropped below 1 after a few steps in Epoch 0, and in some other cases, the loss can keep > 3 for more than 10 epochs. I see the code has `seed` to initialize `ramdon` and `torch`. Could I know if you use any seed to train models in the paper? 

Thank you very much!

![image](https://github.com/user-attachments/assets/1de25763-49d6-4f1a-8205-fc2ff1b67d2f)

Here attached is the args used in my training.
[args.json](https://github.com/user-attachments/files/16242914/args.json)

**Comment by JinhwiPark** [comment]
Hello,

Thank you for your feedback and for testing the model. I apologize for any confusion caused by the discrepancies between the paper and the code.

Regarding model initialization, it indeed plays a crucial role. Unfortunately, the exact seed values used during training have been lost due to log files being deleted from the server. However, I can share the arguments stored on my Weights and Biases (wandb) server that were used during the experiments:

[args (1).json](https://github.com/user-attachments/files/16412547/args.1.json)

Compared to your experimental setup, it appears you did not use the pre-trained weights for the monocular depth estimation model. This is a crucial factor for the model's performance and can significantly aid in initialization. I have uploaded the weights to [Google Drive Link](https://drive.google.com/drive/folders/1kBb-70yvXjIQqGU28RNaHC2aZT1aTy7H), so please use them for better results.
If you have any further questions or need additional assistance, please feel free to reach out.


Thank you!

**Comment by EricYang3721** [comment]
Thanks so much. The performance get much better after including the pretrained model. 

Just one question about `conf_prop`
I saw in your args file, `conf_prop=False`, but in the code, it is set to be `True`. Should we use `True` or `False`? https://github.com/JinhwiPark/DepthPrompting/blob/main/model_list.py#L9C1-L10C1

Thank you very much for the help!

**Comment by JinhwiPark** [comment]
You're welcome! I'm glad to hear that the performance improved with the pretrained model.

Regarding your question about `conf_prop`, you should set it to `True`. This setting ensures that the configuration is correctly aligned with the code implementation for optimal performance.

Thank you for reaching out, and feel free to ask if you have any more questions!
